flusher.go

```diff
@@ -212,6 +214,33 @@ func (s *Server) reportGlobalMetricsFlushCounts(ms metricsSummary) {
 	s.statsd.Count("worker.metrics_flushed_total", int64(ms.totalTimers), []string{"metric_type:timer"}, 1.0)
 }
 
+func (s *Server) flushS3(finalMetrics []DDMetric) {
+	start := time.Now()
+
+	var data bytes.Buffer
+	err := json.NewEncoder(&data).Encode(finalMetrics)
+	if err != nil {
+		s.logger.WithFields(logrus.Fields{
+			"metrics": len(finalMetrics),
+		}).Error("Could not marshal finalMetrics before posting to s3")
+		return
+	}
+
+	// this feels dirty but oh well
+	seekableData := strings.NewReader(data.String())
+
+	err = s3Post(s.Hostname, seekableData)
+	if err != nil {
+		s.logger.WithFields(logrus.Fields{
+			"metrics": len(finalMetrics),
+		}).Error("Error posting to s3")
+		return
+	}
+
+	s.statsd.TimeInMilliseconds("flush.s3.total_duration_ns", float64(time.Now().Sub(start).Nanoseconds()), []string{"part:post"}, 1.0)
+	s.logger.WithField("metrics", len(finalMetrics)).Info("Completed flush to s3")
```

Comment:
nitpick: let's lower this to debug level to reduce log volume, since it's basically repeating the equivalent log line that would be printed from datadog